match/lib/match/runner.rb

```diff
@@ -277,6 +311,10 @@ def fetch_provisioning_profile(params: nil, certificate_id: nil, app_identifier:
         return nil
       end
 
+      if Helper.mac?
```

Comment:
does the fact that we move this after the previous block means that we stop installing expired profiles into the keychain?

Reply:
Yes, I see no point in installing an expired profile, generating a new one, and installing a new one after.

Reply:
jfy, I did the same in my other PR: https://github.com/fastlane/fastlane/pull/21694/files#diff-8cd09d7ab3f4398e3160f603839ac2891e3cdf60e7f09930a1288ce33ab8fc1fL279